src/Middleware/WebSockets/src/HandshakeHelpers.cs

```diff
@@ -72,5 +75,89 @@ public static string CreateResponseKey(string requestKey)
 
             return Convert.ToBase64String(hashedBytes);
         }
+
+        public static bool ParseDeflateOptions(ReadOnlySpan<char> extension, WebSocketDeflateOptions options, [NotNullWhen(true)] out string? response)
+        {
+            response = null;
+            var builder = new StringBuilder(WebSocketDeflateConstants.MaxExtensionLength);
+            builder.Append(WebSocketDeflateConstants.Extension);
+
+            while (true)
+            {
+                int end = extension.IndexOf(';');
+                ReadOnlySpan<char> value = (end >= 0 ? extension[..end] : extension).Trim();
+
+                if (value.Length > 0)
+                {
+                    if (value.SequenceEqual(WebSocketDeflateConstants.ClientNoContextTakeover))
+                    {
+                        // REVIEW: If someone specifies true for server options, do we allow client to override this?
+                        options.ClientContextTakeover = false;
+                        builder.Append("; ").Append(WebSocketDeflateConstants.ClientNoContextTakeover);
+                    }
+                    else if (value.SequenceEqual(WebSocketDeflateConstants.ServerNoContextTakeover))
+                    {
+                        options.ServerContextTakeover = false;
+                        builder.Append("; ").Append(WebSocketDeflateConstants.ServerNoContextTakeover);
```

Comment:
>   A server MAY include the "server_no_context_takeover" extension
   parameter in an extension negotiation response even if the extension
   negotiation offer being accepted by the extension negotiation
   response didn't include the "server_no_context_takeover" extension
   parameter.

Should we add it to the response if ServerContextTakeover = false was set by the server rather than the client?